File: ncRepositorios/pContasPagar.cs
using System;
using System.Collections.Generic;
using Dapper;
using MySql.Data.MySqlClient;
using ncNComum.nsExcecao;
using nsContasPagar;

namespace ncPersistencia.nsContasPagar
{
    public class pContasPagar : RepositorioBase, IpContasPagar
    {
        public ColecaoContasPagar Listar()
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var lista = conn.Query<dContasPagar>(
                        "SELECT cid, descricao, valor, dataEmissao, dataVencimento, dataPagamento, situacao, fornecedor FROM ContasPagar").AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoContasPagar();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Listar ContasPagar [" + ToString() + "] - " + ex.Message);
            }
        }

        public ColecaoContasPagar Consultar(dContasPagar dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    var conditions = new List<string>();
                    var p = new DynamicParameters();
                    if (dados.cid != null && dados.cid != 0) { conditions.Add("cid=@cid"); p.Add("cid", dados.cid); }
                    if (!string.IsNullOrEmpty(dados.observacao)) { conditions.Add("descricao=@descricao"); p.Add("descricao", dados.observacao); }
                    if (dados.valor != null && dados.valor != 0) { conditions.Add("valor=@valor"); p.Add("valor", dados.valor); }
                    if (!string.IsNullOrEmpty(dados.dataEmissao)) { conditions.Add("dataEmissao=@dataEmissao"); p.Add("dataEmissao", dados.dataEmissao); }
                    if (!string.IsNullOrEmpty(dados.dataVencimento)) { conditions.Add("dataVencimento=@dataVencimento"); p.Add("dataVencimento", dados.dataVencimento); }
                    if (!string.IsNullOrEmpty(dados.dataPagamento)) { conditions.Add("dataPagamento=@dataPagamento"); p.Add("dataPagamento", dados.dataPagamento); }
                    if (dados.pago.HasValue && dados.pago.Value)
                    {
                        conditions.Add("situacao=@situacao");
                        p.Add("situacao", dados.pago);
                    }
                    if (dados.fornecedor_cid != null && dados.fornecedor_cid != 0)
                    {
                        conditions.Add("fornecedor=@fornecedor");
                        p.Add("fornecedor", dados.fornecedor_cid);
                    }
                    var where = conditions.Count > 0 ? "WHERE " + string.Join(" AND ", conditions) : "";
                    var sql = $"SELECT cid, descricao, valor, dataEmissao, dataVencimento, dataPagamento, situacao, fornecedor FROM ContasPagar {where}";
                    var lista = conn.Query<dContasPagar>(sql, p).AsList();
                    if (lista.Count == 0) return null;
                    var retorno = new ColecaoContasPagar();
                    retorno.AddRange(lista);
                    return retorno;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Consultar ContasPagar [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Incluir(dContasPagar dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    conn.Execute(
                        "INSERT INTO ContasPagar (descricao, valor, dataEmissao, dataVencimento, dataPagamento, situacao, fornecedor) " +
                        "VALUES (@descricao, @valor, @dataEmissao, @dataVencimento, @dataPagamento, @situacao, @fornecedor)",
                        new {
                            descricao = dados.observacao, valor = dados.valor, dataEmissao = dados.dataEmissao,
                            dataVencimento = dados.dataVencimento, dataPagamento = dados.dataPagamento,
                            situacao = dados.pago, fornecedor = dados.fornecedor_cid
                        });
                    return (int)conn.ExecuteScalar<long>("SELECT LAST_INSERT_ID()");
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Incluir ContasPagar [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Importar(dContasPagar dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    conn.Execute(
                        "INSERT INTO ContasPagar (cid, descricao, valor, dataEmissao, dataVencimento, dataPagamento, situacao, fornecedor) " +
                        "VALUES (@cid, @descricao, @valor, @dataEmissao, @dataVencimento, @dataPagamento, @situacao, @fornecedor)",
                        new {
                            cid = dados.cid, descricao = dados.observacao, valor = dados.valor,
                            dataEmissao = dados.dataEmissao, dataVencimento = dados.dataVencimento,
                            dataPagamento = dados.dataPagamento, situacao = dados.pago, fornecedor = dados.fornecedor_cid
                        });
                    return 1;
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Importar ContasPagar [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Alterar(dContasPagar dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute(
                        "UPDATE ContasPagar SET descricao=@descricao, valor=@valor, dataEmissao=@dataEmissao, " +
                        "dataVencimento=@dataVencimento, dataPagamento=@dataPagamento, situacao=@situacao, fornecedor=@fornecedor " +
                        "WHERE cid=@cid",
                        new {
                            descricao = dados.observacao, valor = dados.valor, dataEmissao = dados.dataEmissao,
                            dataVencimento = dados.dataVencimento, dataPagamento = dados.dataPagamento,
                            situacao = dados.pago, fornecedor = dados.fornecedor_cid, cid = dados.cid
                        });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Alterar ContasPagar [" + ToString() + "] - " + ex.Message);
            }
        }

        public int Excluir(dContasPagar dados)
        {
            try
            {
                using (var conn = CriarConexao())
                {
                    return conn.Execute("DELETE FROM ContasPagar WHERE cid=@cid", new { cid = dados.cid });
                }
            }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex)
            {
                throw new ExcecaoNascomercio("Erro em Excluir ContasPagar [" + ToString() + "] - " + ex.Message);
            }
        }
    }
}

File: ncPersistencia/pContasPagar.vb
Imports ncDados.nsContasPagar
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes
Imports ncComum.nsExcecao

Namespace nsContasPagar

  Public Class pContasPagar

    Public Function Listar() As ColecaoContasPagar

      Dim retorno As ColecaoContasPagar
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dContasPagar
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " Select cid, codigo, codigoBarra, valor, aceite, observacao, " & _
                    " dataEmissao, dataVencimento, dataPagamento, valorPagamento, fornecedor_cid, pago " & _
                    " From ContasPagar "

        ds = acessoBanco.ExecutarDS(comandoSQL)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoContasPagar

              For Each row In dt.Rows
                item = New dContasPagar

                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                item.codigo = cFuncoes.RetornarTexto(row("codigo"))
                item.valor = cFuncoes.RetornarDecimal(row("valor"))
                item.observacao = cFuncoes.RetornarTexto(row("observacao"))
                item.aceite = cFuncoes.RetornarBoleano(row("aceite"))
                item.dataEmissao = cFuncoes.RetornarTexto(row("dataEmissao"))
                item.dataVencimento = cFuncoes.RetornarTexto(row("dataVencimento"))
                item.dataPagamento = cFuncoes.RetornarTexto(row("dataPagamento"))
                item.valorPagamento = cFuncoes.RetornarDecimal(row("valorPagamento"))
                item.codigoBarra = cFuncoes.RetornarTexto(row("codigoBarra"))
                item.fornecedor_cid = cFuncoes.RetornarInteiro(row("fornecedor_cid"))
                item.pago = cFuncoes.RetornarBoleano(row("pago"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Listar ContasPagar [" & Me.ToString() & "] - " & ex.Message, ex)

      End Try

      Listar = retorno

    End Function

    Public Function Consultar(ByVal dados As dContasPagar) As ColecaoContasPagar

      Dim retorno As ColecaoContasPagar
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dContasPagar
      Dim sqlSelect As String
      Dim sqlWhere As String
      Dim sqlFrom As String

      Try

        acessoBanco = New cAcessoBD

        sqlSelect = " Select cid, codigo, codigoBarra, valor, aceite, observacao, " & _
                    " dataEmissao, dataVencimento, dataPagamento, valorPagamento, fornecedor_cid, pago "
        sqlWhere = String.Empty
        sqlFrom = " From ContasPagar "

        '-- cid
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.cid, "cid")
        '-- codigo
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.codigo, "codigo")
        '-- codigoBarra
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.codigoBarra, "codigoBarra")
        '-- valor
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.valor, "valor")
        '-- aceite
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.aceite, "aceite")
        '-- observacao
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.observacao, "observacao")
        '-- dataEmissao
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.dataEmissao, "dataEmissao")
        '-- dataPagamento
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.dataPagamento, "dataPagamento")
        '-- dataVencimento
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.dataVencimento, "dataVencimento")
        '-- valorPagamento
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.valorPagamento, "valorPagamento")
        '-- fornecedor_cid
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.fornecedor_cid, "fornecedor_cid")
        '-- aceite
        sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.pago, "pago")

        If Not sqlWhere.Equals(String.Empty) Then
          sqlWhere = " WHERE " & sqlWhere
        End If

        ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoContasPagar

              For Each row In dt.Rows
                item = New dContasPagar

                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                item.codigo = cFuncoes.RetornarTexto(row("codigo"))
                item.valor = cFuncoes.RetornarDecimal(row("valor"))
                item.observacao = cFuncoes.RetornarTexto(row("observacao"))
                item.aceite = cFuncoes.RetornarBoleano(row("aceite"))
                item.dataEmissao = cFuncoes.RetornarTexto(row("dataEmissao"))
                item.dataVencimento = cFuncoes.RetornarTexto(row("dataVencimento"))
                item.dataPagamento = cFuncoes.RetornarTexto(row("dataPagamento"))
                item.valorPagamento = cFuncoes.RetornarDecimal(row("valorPagamento"))
                item.codigoBarra = cFuncoes.RetornarTexto(row("codigoBarra"))
                item.fornecedor_cid = cFuncoes.RetornarInteiro(row("fornecedor_cid"))
                item.pago = cFuncoes.RetornarBoleano(row("pago"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Consultar ContasPagar [" & Me.ToString() & "] - " & ex.Message, ex)

      End Try

      Consultar = retorno

    End Function

    Public Function Incluir(ByVal dados As dContasPagar) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " INSERT INTO " & _
            " ContasPagar ( codigo, codigoBarra, valor, aceite, observacao, " & _
            " dataEmissao, dataVencimento, dataPagamento, valorPagamento, fornecedor_cid, pago ) " & _
            " VALUES (" & _
            cFuncoes.PersistirTexto(dados.codigo) & "," & _
            cFuncoes.PersistirTexto(dados.codigoBarra) & "," & _
            cFuncoes.PersistirDecimal(dados.valor) & "," & _
            cFuncoes.PersistirBoleano(dados.aceite) & "," & _
            cFuncoes.PersistirTexto(dados.observacao) & "," & _
            cFuncoes.PersistirData(dados.dataEmissao) & "," & _
            cFuncoes.PersistirData(dados.dataVencimento) & "," & _
            cFuncoes.PersistirData(dados.dataPagamento) & "," & _
            cFuncoes.PersistirDecimal(dados.valorPagamento) & "," & _
            cFuncoes.PersistirInteiro(dados.fornecedor_cid) & "," & _
            cFuncoes.PersistirBoleano(dados.pago) & ")"

        retorno = acessoBanco.ExecutarCID(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Incluir ContasPagar [" & Me.ToString() & "] - " & ex.Message, ex)

      End Try

      Incluir = retorno

    End Function

    Public Function Importar(ByVal dados As dContasPagar) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " INSERT INTO " & _
            " ContasPagar ( cid, codigo, codigoBarra, valor, aceite, observacao, " & _
            " dataEmissao, dataVencimento, dataPagamento, valorPagamento, fornecedor_cid, pago ) " & _
            " VALUES (" & _
            cFuncoes.PersistirInteiro(dados.cid) & _
            cFuncoes.PersistirTexto(dados.codigo) & _
            cFuncoes.PersistirTexto(dados.codigoBarra) & _
            cFuncoes.PersistirDecimal(dados.valor) & _
            cFuncoes.PersistirInteiro(dados.aceite) & _
            cFuncoes.PersistirTexto(dados.observacao) & _
            cFuncoes.PersistirData(dados.dataEmissao) & _
            cFuncoes.PersistirData(dados.dataVencimento) & _
            cFuncoes.PersistirData(dados.dataPagamento) & _
            cFuncoes.PersistirDecimal(dados.valorPagamento) & _
            cFuncoes.PersistirInteiro(dados.fornecedor_cid) & _
            cFuncoes.PersistirInteiro(dados.pago) & ")"

        retorno = acessoBanco.ExecutarCID(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Importar ContasPagar [" & Me.ToString() & "] - " & ex.Message, ex)

      End Try

      Importar = retorno

    End Function

    Public Function Alterar(ByVal dados As dContasPagar) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " UPDATE ContasPagar SET " & _
            " codigo = " & cFuncoes.PersistirTexto(dados.codigo) & "," & _
            " codigoBarra = " & cFuncoes.PersistirTexto(dados.codigoBarra) & "," & _
            " valor = " & cFuncoes.PersistirDecimal(dados.valor) & "," & _
            " aceite = " & cFuncoes.PersistirBoleano(dados.aceite) & "," & _
            " observacao = " & cFuncoes.PersistirTexto(dados.observacao) & "," & _
            " dataEmissao = " & cFuncoes.PersistirData(dados.dataEmissao) & "," & _
            " dataVencimento = " & cFuncoes.PersistirData(dados.dataVencimento) & "," & _
            " dataPagamento = " & cFuncoes.PersistirData(dados.dataPagamento) & "," & _
            " valorPagamento = " & cFuncoes.PersistirDecimal(dados.valorPagamento) & "," & _
            " fornecedor_cid = " & cFuncoes.PersistirInteiro(dados.fornecedor_cid) & "," & _
            " pago = " & cFuncoes.PersistirBoleano(dados.pago) & _
            " WHERE " & _
            " cid = " & dados.cid.ToString()

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Alterar ContasPagar [" & Me.ToString() & "] - " & ex.Message, ex)

      End Try

      Alterar = retorno

    End Function

    Public Function Excluir(ByVal dados As dContasPagar) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " DELETE FROM ContasPagar " & _
            " WHERE cid = " & dados.cid.ToString()

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Excluir ContasPagar [" & Me.ToString() & "] - " & ex.Message, ex)

      End Try

      Excluir = retorno

    End Function

  End Class

End Namespace